aries_cloudagent/protocols/out_of_band/v1_0/messages/invitation.py

```diff
@@ -143,20 +143,17 @@ def __init__(
         )
         self.requests_attach = list(requests_attach) if requests_attach else []
         self.services = services
-        self.assign_version_to_message_type(version=version)
+        if msg_type:
+            self._type = msg_type
+        else:
+            self._type = self.get_updated_msg_type(version)
```

Comment:
Maybe the AgentMessage constructor can accept a couple more options to avoid reassigning the message type and repeating this code. For example:

```super().__init__(_message_type=msg_type, _message_version=version, **kwargs)```

If `_message_type` is null then it defaults to `Meta.message_type`. If `_message_version` is non-null then it performs the substitution.

Reply:
(Or just `_type` and `_version`)

Reply:
Something like below in `AgentMessage -> __init__`:
```
if _type:
    self._message_type = _type
elif _version:
    self._message_type = self.get_updated_msg_type(_version)
else:
    self._message_type = self.Meta.message_type
```

Reply:
Yes I think that would work